File: Cpp/Rotate Function.cpp
// 4ms, 100.0%
static int x = []() {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);
	return 0;
}();

class Solution
{
public:
	int maxRotateFunction(const vector<int>& A)
	{
		int res = 0, sum = accumulate(A.begin(), A.end(), 0), size = A.size();
		for(int i = 0; i < size; ++i) res += i * A[i];
		int tmp = res;
		for(int i = 0; i < size - 1; ++i) {
			tmp = tmp - sum + size * A[i];
			res = max(res, tmp);
		}
		return res;
	}
};

File: Python3/Rotate Function.py
# 88ms, 18.52%
class Solution:
    def maxRotateFunction(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        res, sums, size = 0, sum(A), len(A)
        for i in range(size):
            res += i * A[i]
        tmp = res
        for i in range(size - 1):
            tmp = tmp - sums + size * A[i]
            res = max(res, tmp)
        return res
